EventFilter/L1TRawToDigi/src/implementations_stage2/MPUnpacker_0x1001000b.cc

```diff
@@ -0,0 +1,229 @@
+     switch(block.header().getID()){
+     case 123:  ht.setType(l1t::EtSum::kTotalHt);  break;
+     case 121:  ht.setType(l1t::EtSum::kTotalHtx); break;
+     case 127:  ht.setType(l1t::EtSum::kTotalHty); break;
+     case 125:  ht.setType(l1t::EtSum::kTotalHt);  break;
+     case 131:  ht.setType(l1t::EtSum::kTotalHtx); break;
+     case 129: ht.setType(l1t::EtSum::kTotalHty); break;
+     default: break;
+     }
+
+     LogDebug("L1T") << "HT/MHTx/MHTy: pT " << ht.hwPt();
+
+     res2_->push_back(0,ht);
+
+     // Two jets
+     for (unsigned nJet=0; nJet < 2; nJet++){
+
+       raw_data = block.payload()[fjet+nJet];
+
+       if (raw_data == 0)
+            continue;
+
+       l1t::Jet jet = l1t::Jet();
+
+       int etasign = 1;
+       if ((block.header().getID() == 125) ||
+           (block.header().getID() == 131) ||
+           (block.header().getID() == 129)) {
+         etasign = -1;
+       }
+
+       LogDebug("L1") << "block ID=" << block.header().getID() << " etasign=" << etasign;
+
+       int mpEta = etasign*(raw_data & 0x3F);
+       jet.setHwEta(CaloTools::caloEta(mpEta));
+       jet.setHwPhi((raw_data >> 6) & 0x7F);
+       jet.setHwPt((raw_data >> 13) & 0xFFFF);
+       jet.setHwQual((raw_data>>29) & 0x1 );
+         
+       LogDebug("L1T") << "Jet: eta " << jet.hwEta() << " phi " << jet.hwPhi() << " pT " << jet.hwPt() << " qual " << jet.hwQual();
+
+       jet.setP4( l1t::CaloTools::p4MP(&jet) );
+       res1_->push_back(0,jet);
+
+     }
+
+     //      ===== EGammas =====
+     
+     // Two EGammas
+
+
+     for (unsigned nEG=0; nEG < 2; nEG++){
+       
+       raw_data = block.payload()[feg+nEG];
+       
+       if (raw_data == 0)
+	 continue;
+       
+       l1t::EGamma eg = l1t::EGamma();
+
+       int etasign = 1;
+       if ((block.header().getID() == 125) ||
+           (block.header().getID() == 131) ||
+           (block.header().getID() == 129)) {
+         etasign = -1;
+       }
+       
+       LogDebug("L1") << "block ID=" << block.header().getID() << " etasign=" << etasign;
+
+       eg.setHwEta(etasign*((raw_data >> 3) & 0x3F));
+       eg.setHwPhi((raw_data >> 9) & 0x7F);
+       eg.setHwPt((raw_data >> 20) & 0xFFF);
+       eg.setHwIso((raw_data>>1) & 0x3);       
+       eg.setHwQual((raw_data>>16) & 0xf );
+
+	   
+       LogDebug("L1T") << "Egamma: eta " << eg.hwEta() << " phi " << eg.hwPhi() << " pT " << eg.hwPt() << " qual " << eg.hwQual();
+       
+       eg.setP4( l1t::CaloTools::p4MP(&eg) );
+       res3_->push_back(0,eg);
+     }
+
+     
+     //      ===== Taus =====
+     
+     // Two taus
+
+     for (unsigned nTau=0; nTau < 2; nTau++){
+       
+       raw_data = block.payload()[ftau+nTau];
+       
+       if (raw_data == 0)
+	 continue;
+       
+       l1t::Tau tau = l1t::Tau();
+
+       int etasign = 1;
+       if ((block.header().getID() == 125) ||
+           (block.header().getID() == 131) ||
+           (block.header().getID() == 129)) {
+         etasign = -1;
```

Comment:
Magic numbers are back. Who knows what `123` is? `0xFFFFFF ` this could be a macro with a human readable name (SOMETHING_MASK). Here: `<< 16`, that `16` can also be a human readable macro (SOMETHING_SHIFT). Usually naming should match to what you have in official specification/datasheet. It wouldn't hurt adding some comments explaining raw data format either, or/and a document (text file) with layouts, specs, etc. It helps __a lot__ when someone needs to look into this.